pkg/proto/pbgo/trace/idx/internal_span.go

```diff
@@ -0,0 +1,148 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package idx
+
+// StringTable is a table of strings that is used to store the de-duplicated strings in a trace
+type StringTable struct {
+	strings []string
+	lookup  map[string]uint32
+}
+
+// NewStringTable creates a new string table, always starts with an empty string
+func NewStringTable() *StringTable {
+	return &StringTable{
+		strings: []string{""},
+		lookup:  map[string]uint32{"": 0},
+	}
+}
+
+// addUnchecked adds a string to the string table without checking for duplicates
+func (s *StringTable) addUnchecked(str string) uint32 {
+	s.strings = append(s.strings, str)
+	s.lookup[str] = uint32(len(s.strings) - 1)
+	return uint32(len(s.strings) - 1)
+}
+
+// Add adds a string to the string table if it doesn't already exist and returns the index of the string
+func (s *StringTable) Add(str string) uint32 {
+	if idx, ok := s.lookup[str]; ok {
+		return idx
+	}
+	return s.addUnchecked(str)
+}
+
+// Get returns the string at the given index - panics if out of bounds
```

Comment:
to ease the access, maybe the signature and implementation could account for missing strings by returning `(string, error)` or `(string, bool)`?

Reply:
Yeah I think that's a good idea - I think most usages outside this package will not call this method directly, but use higher level functions that handle interacting with the string table for you